client/web/src/enterprise/executors/instances/ExecutorNode.tsx

```diff
@@ -42,16 +42,28 @@ export const ExecutorNode: FunctionComponent<React.PropsWithChildren<ExecutorNod
                                 </Tooltip>
                             )}
                             {node.hostname}{' '}
-                            <Badge
-                                variant="secondary"
-                                tooltip={`The executor is configured to pull data from the queue "${node.queueName}"`}
-                            >
-                                {node.queueName}
-                            </Badge>
+                            {node.queueName !== null &&
+                                <Badge
+                                    variant="secondary"
+                                    tooltip={`The executor is configured to pull data from the queue "${node.queueName}"`}
+                                >
+                                    {node.queueName}
+                                </Badge>
+                            }
+                            {(node.queueNames as string[])?.map(queue =>
```

Comment:
Since `queueNames` is resolved to `nil` if there's `e.executor.QueueNames` has length 0, `node.queueNames` could be `null` here. This is probably safer and a bit more idiomatic, then:

 ```suggestion
                            {(node.queueNames || []).map(queue =>
```

If it's `null`, it'll still not render any `Badge`s. Using `as T` assertions is generally discouraged since it overrides the type that is directly generated from the GraphQL schema and introduces the risk of disagreeing types if that GraphQL schema type were ever to change (e.g. if it became an enum type or a nested object type in the future)

Reply:
Good to know 🙂 

My IDE reports these warnings when implementing your suggestion:
```
ESLint: Unsafe call of an `any` typed value.(@typescript-eslint/no-unsafe-call)
ESLint: Unsafe member access .map on an `any` value.(@typescript-eslint/no-unsafe-member-access)
```

Are these safe to ignore?

Reply:
I think that's likely because the TS language server hasn't picked up the updated type for `node` that includes `queueNames` yet. The TS language server in most IDEs seems to struggle to pick up changes in the auto-generated GraphQL types, unfortunately. If you open the file `client/shared/src/graphql-operations.ts` in your IDE and wait a moment, does the warning go away? Or if not, what if you run `pnpm generate` to re-generate the file? Or as a last resort, reloading your IDE should *definitely* pick it up. Sorry it's so finnicky.